[Bug] [Athena Datasource]  datasource other params illegal

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

error log
```
[ERROR] 2022-11-28 07:37:52.705 +0000 org.apache.dolphinscheduler.api.exceptions.ApiExceptionHandler:[53] - 建立数据源连接失败
java.lang.IllegalArgumentException: datasource other params illegal
        at org.apache.dolphinscheduler.plugin.datasource.api.datasource.AbstractDataSourceProcessor.checkOther(AbstractDataSourceProcessor.java:89)
        at org.apache.dolphinscheduler.plugin.datasource.api.datasource.AbstractDataSourceProcessor.checkDatasourceParam(AbstractDataSourceProcessor.java:50)
        at org.apache.dolphinscheduler.plugin.datasource.api.utils.DataSourceUtils.checkDatasourceParam(DataSourceUtils.java:47)
        at org.apache.dolphinscheduler.api.controller.DataSourceController.connectDataSource(DataSourceController.java:218)
        at org.apache.dolphinscheduler.api.controller.DataSourceController$$FastClassBySpringCGLIB$$835fdd04.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)

```


![00](https://user-images.githubusercontent.com/1808782/204225330-ec47556d-5dc3-46d5-9159-3312399f7b37.png)




### What you expected to happen

Add athena  datasource  normally  

### How to reproduce

when  add athena datasource .
set other jdbc params.
```
{"s3_staging_dir":"s3://aws-athena-results-ap-south-1/all-logs"}
```


### Anything else

_No response_

### Version

3.1.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/datasource/AbstractDataSourceProcessor.java
+++ b/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/datasource/AbstractDataSourceProcessor.java
@@ -39,7 +39,7 @@ public abstract class AbstractDataSourceProcessor implements DataSourceProcessor
 
     private static final Pattern DATABASE_PATTER = Pattern.compile("^[a-zA-Z0-9\\_\\-\\.]+$");
 
-    private static final Pattern PARAMS_PATTER = Pattern.compile("^[a-zA-Z0-9\\-\\_\\/\\@\\.]+$");
+    private static final Pattern PARAMS_PATTER = Pattern.compile("^[a-zA-Z0-9\\-\\_\\/\\@\\.\\:]+$");
 
     private static final Set<String> POSSIBLE_MALICIOUS_KEYS = Sets.newHashSet("allowLoadLocalInfile");
 

--- a/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/test/java/org/apache/dolphinscheduler/plugin/datasource/api/datasource/AbstractDataSourceProcessorTest.java
+++ b/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/test/java/org/apache/dolphinscheduler/plugin/datasource/api/datasource/AbstractDataSourceProcessorTest.java
@@ -49,6 +49,7 @@ public class AbstractDataSourceProcessorTest {
         AbstractDataSourceProcessor mockDataSourceProcessor = mock(AbstractDataSourceProcessor.class);
         Map<String, String> other = new HashMap<>();
         other.put("allowLoadLocalInfile", "whatever");
+        other.put("OutputLocation", "s3://test-data-xini/athena-query");
         doThrow(new IllegalArgumentException()).when(mockDataSourceProcessor).checkOther(other);
     }
 }

